pinot-query-planner/src/main/java/org/apache/pinot/query/planner/nodes/TableScanNode.java

```diff
@@ -0,0 +1,57 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.query.planner.nodes;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.apache.calcite.rel.logical.LogicalTableScan;
+import org.apache.calcite.rel.type.RelDataTypeField;
+
+
+public class TableScanNode extends AbstractStageNode {
+  private final List<String> _tableName;
+  private final List<String> _tableScanColumns;
+
+  public TableScanNode(LogicalTableScan tableScan, String stageId) {
+    super(stageId);
+    _tableName = tableScan.getTable().getQualifiedName();
+    // TODO: optimize this, table field is not directly usable as name.
+    _tableScanColumns =
+        tableScan.getRowType().getFieldList().stream().map(RelDataTypeField::getName).collect(Collectors.toList());
+  }
+
+  @Override
+  public List<StageNode> getInputs() {
+    return Collections.emptyList();
+  }
+
+  @Override
+  public void addInput(StageNode queryStageRoot) {
+    throw new UnsupportedOperationException("table scan cannot add input");
```

Comment:
(nit) suggest changing the message to `TableScanNode cannot add input as it is a leaf node` ?